cmd/podman/machine/list.go

```diff
@@ -115,37 +111,29 @@ func outputTemplate(cmd *cobra.Command, responses []*entities.ListReporter) erro
 		"Memory":   "MEMORY",
 		"DiskSize": "DISK SIZE",
 	})
-	printHeader := !listFlag.noHeading
-	if listFlag.quiet {
-		printHeader = false
-	}
-	var row string
+
+	rpt := report.New(os.Stdout, cmd.Name())
+	defer rpt.Flush()
+
+	var err error
 	switch {
-	case cmd.Flags().Changed("format"):
-		row = cmd.Flag("format").Value.String()
-		printHeader = report.HasTable(row)
-		row = report.NormalizeFormat(row)
+	case cmd.Flag("format").Changed:
```

Comment:
FYI `cmd.Flags().Changed("format")` guards against using a nil pointer while `cmd.Flag("format").Changed` does not. Only exposed when flag is undefined.

Reply:
I don't think that matters much since a) we know that the flag exists and b) this PR includes a test for *all* format flags so if it would panic the test will catch it.